src/status_im2/contexts/wallet/select_address_to_watch/view.cljs

```diff
@@ -1,27 +1,59 @@
 (ns status-im2.contexts.wallet.select-address-to-watch.view
   (:require
-    [clojure.string :as string]
+    clojure.string
     [quo.core :as quo]
-    [quo.theme :as quo.theme]
     [re-frame.core :as re-frame]
-    [react-native.clipboard :as clipboard]
     [react-native.core :as rn]
     [react-native.safe-area :as safe-area]
     [reagent.core :as reagent]
+    [status-im2.constants :as constants]
     [status-im2.contexts.wallet.select-address-to-watch.style :as style]
+    [utils.debounce :as debounce]
     [utils.i18n :as i18n]
     [utils.re-frame :as rf]))
 
-(defn view-internal
+(defn- address-input
+  [input-value input-focused?]
```

Comment:
similar code to send scan address, most pieces here are reusable so I just copied the relevant pieces over to avoid coupling the send and watch only scanners.